src/rust/uk.mangainua/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+	let mut manga_arr: Vec<Manga> = Vec::new();
+	let mut total: i32 = 1;
+
+	let base_url = String::from("https://manga.in.ua");
+
+	let genres_list = helper::genres_list();
+
+	let mut sort_value: String = String::new();
+	let mut search_value = String::new();
+	let mut genre_value = String::new();
+	let mut status_value : String = String::new();
+
+	let mut is_cover_data_src = false;
+
+	for filter in filters {
+		match filter.kind{
+			FilterType::Title => {
+				search_value = filter.value.as_string()?.read();
+			}
+			FilterType::Select => {
+				if filter.name.as_str() == "Сортувати" {
+					let index = filter.value.as_int()? as usize;
+
+					let option = match index {
+						0 => "latest",
+						1 => "popular",
+						_ => "",
+					};
+					sort_value = String::from(option);
+				}
+
+				if filter.name.as_str() == "Жанри" {
+					let index = filter.value.as_int()? as usize;
+					match index {
+						0 => continue,
+						_ => genre_value = String::from(genres_list[index]),
+					}
+				}
+
+				if filter.name.as_str() == "Статус перекладу" {
+					let index = filter.value.as_int()? as usize;
+
+					let option = match index {
+						0 => continue,
+						1 => "xfsearch/tra/%D0%97%D0%B0%D0%BA%D1%96%D0%BD%D1%87%D0%B5%D0%BD%D0%B8%D0%B9", // Закінчений - Completed
+						2 => "xfsearch/tra/%D0%A2%D1%80%D0%B8%D0%B2%D0%B0%D1%94", // Триває - Ongoing
+						3 => "xfsearch/tra/%D0%9D%D0%B5%D0%B2%D1%96%D0%B4%D0%BE%D0%BC%D0%BE", // Невідомо - Unknown 
+						4 => "xfsearch/tra/%D0%9F%D0%BE%D0%BA%D0%B8%D0%BD%D1%83%D1%82%D0%BE", // Покинуто - Cancelled
+						_ => continue
+					};
+					status_value = String::from(option);
+				}
+			}
+			_ => todo!()
+		}
+	}
+
+	// in case the user searches on popular - move to latest because popular doesn't have a search
+	if sort_value == "popular" && (!search_value.is_empty() || !genre_value.is_empty() || !status_value.is_empty()) {
+		sort_value = String::from("latest");
+	}
+
+	if sort_value == "popular" {
+		// ignore page number
+		let html = Request::new("https://manga.in.ua/", HttpMethod::Get).html();
+
+		for result in html.select(".owl-carousel .card--big").array() {
+			let res_node = result.as_node();
+
+			let href = res_node.select("a").attr("href").read();
+
+			let title = res_node.select(".card__content .card__title a").text().read();
+			let cover = res_node.select(".card__cover a figure img").attr("abs:src").read(); 
+
+			let mut is_nsfw : MangaContentRating = MangaContentRating::Safe;
+			let mut categories: Vec<String> = Vec::new();
+			for categ_res in res_node.select(".card__category a").array(){
+				let categ_node = categ_res.as_node();
+				let name = categ_node.text().read(); 
+				is_nsfw = helper::is_nsfw(name.clone());
+
+				categories.push(name);
+			}
+
+			manga_arr.push(Manga{
+				id: href,
+				cover,
+				title,
+				author: String::new(),
+				artist: String::new(),
+				description: String::new(),
+				url: String::new(),
```

Comment:
instead of providing empty strings, do `..Default::default()` at the end

Reply:
How can I fix it?
error[E0277]: the trait bound `aidoku::Manga: core::default::Default` is not satisfied
   --> src/lib.rs:289:5
    |
289 |         ..Default::default()
    |           ^^^^^^^^^^^^^^^^ the trait `core::default::Default` is not implemented for `aidoku::Manga`